### PR TITLE
add target Stellar F4 V2

### PR DESCRIPTION
**Pull request by cvetaevvitaliy**
* add target Stellar F4 V2

### CODE REVIEW

File: configs/STELLARF4V2/config.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+#define USE_FLASH_M25P16
+#define USE_FLASH_W25N01G
+#define USE_FLASH_W25M02G
+
+#define BEEPER_PIN          PB2
+#define MOTOR1_PIN          PC9 // TIM8 CH4 (AF3) DMA2 Stream 7 Channel 7
+#define MOTOR2_PIN          PC8 // TIM8 CH3 (AF3) DMA2 Stream 4 Channel 7
+#define MOTOR3_PIN          PC7 // TIM8 CH2 (AF3) DMA2 Stream 3 Channel 7
+#define MOTOR4_PIN          PC6 // TIM8 CH1 (AF3) DMA2 Stream 2 Channel 7
+#define MOTOR5_PIN          PB6 // or SERVO TIM4 CH1 (AF2)
```
Comment (haslinghuis) [inline configs/STELLARF4V2/config.h:53]: See https://betaflight.com/docs/development/manufacturer/config-target-guidance

File: configs/STELLARF4V2/config.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+#define USE_ACC_SPI_ICM42688P
+#define USE_GYRO
+#define USE_GYRO_SPI_ICM42605
+#define USE_GYRO_SPI_ICM42688P
+#define USE_GYRO_CLKIN
+#define USE_BARO
+#define USE_BARO_BMP280
+#define USE_BARO_DPS310
+#define USE_MAX7456
+#define USE_SDCARD
```
Comment (haslinghuis) [inline configs/STELLARF4V2/config.h:42]: Not used
```suggestion
```

File: configs/STELLARF4V2/config.h [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define FC_TARGET_MCU STM32F405
+#define BOARD_NAME STELLARF4
```
Comment (haslinghuis) [inline configs/STELLARF4V2/config.h:25]:
```suggestion
#define BOARD_NAME STELLARF4V2
```

File: configs/STELLARF4V2/config.h [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+#define USE_GYRO_SPI_ICM42688P
+#define USE_GYRO_CLKIN
+#define USE_BARO
+#define USE_BARO_BMP280
+#define USE_BARO_DPS310
+#define USE_MAX7456
+#define USE_SDCARD
+#define USE_FLASH
+#define USE_FLASH_M25P16
+#define USE_FLASH_W25N01G
```
Comment (haslinghuis) [inline configs/STELLARF4V2/config.h:45]: W25N02G relies on W25N01G
```suggestion
```

File: configs/STELLARF4V2/config.h (hunk 2 above)
Comment (cvetaevvitaliy) [reply configs/STELLARF4V2/config.h]: I was thinking about the future, you can use SD NAND in an LGA8 package

https://www.lcsc.com/datasheet/lcsc_datasheet_2304140030_XTX-XTSD04GLGEAG_C558839.pdf

I'd better check it in the prototype fits.

**Comment by sugaarK** [comment]
unfortunately @cvetaevvitaliy  no more STM f4 or f7 targets with more than 4 motors
as of yesterday.. it was post on our FB page, discord announce and manufacture announcement channels

**Comment by cvetaevvitaliy** [comment]
> unfortunately @cvetaevvitaliy  no more STM f4 or f7 targets with more than 4 motors
> 
> as of yesterday.. it was post on our FB page, discord announce and manufacture announcement channels 
> 
> 

I worked so hard when I was developing that 👨‍💻 
I didn't even have time to watch the news...

4 motors only for F4&F7

**Comment by sugaarK** [comment]
yup.

**Comment by cvetaevvitaliy** [comment]
> yup.
> 
> 

Target has only 4 motors, other pins - servos, UART1(tx rx)

**Review by haslinghuis — APPROVED** [review]

File: configs/STELLARF4V2/config.h [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                            TIMER_PIN_MAP( 5,  PB7,  1, -1 ) \
+                            TIMER_PIN_MAP( 6,  PB8,  1, -1 ) \
+                            TIMER_PIN_MAP( 7,  PB9,  1, -1 ) \
+                            TIMER_PIN_MAP( 8,  PA9,  1, -1 ) \
+                            TIMER_PIN_MAP( 9,  PA10, 1, -1 ) \
+                            TIMER_PIN_MAP( 10, PA8,  1, -1 ) \
+                            TIMER_PIN_MAP( 11, PA3 , 2,  0 ) \
+                            TIMER_PIN_MAP( 12, PB0,  2, -1 )
+
+#define ADC1_DMA_OPT 0 // DMA2 Stream 0 Channel 0
```
Comment (ot0tot) [inline configs/STELLARF4V2/config.h:102]: Need to use ADC3, as ADC1 on Stream 0 conflicts with SPI1_RX and ADC1 on Stream 4 conflicts with TIM8_CH3.

File: configs/STELLARF4V2/config.h [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+#define SERIALRX_UART SERIAL_PORT_USART2
+#define SERIALRX_PROVIDER SERIALRX_CRSF
+
+#define MAG_I2C_INSTANCE (I2CDEV_2)
+#define BARO_I2C_INSTANCE (I2CDEV_2)
+#define DASHBOARD_I2C_INSTANCE (I2CDEV_2)
```
Comment (ot0tot) [inline configs/STELLARF4V2/config.h:109]:
```suggestion

```

File: configs/STELLARF4V2/config.h [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                            TIMER_PIN_MAP( 11, PA3 , 2,  0 ) \
+                            TIMER_PIN_MAP( 12, PB0,  2, -1 )
+
+#define ADC1_DMA_OPT 0 // DMA2 Stream 0 Channel 0
+
+#define SERIALRX_UART SERIAL_PORT_USART2
+#define SERIALRX_PROVIDER SERIALRX_CRSF
+
+#define MAG_I2C_INSTANCE (I2CDEV_2)
+#define BARO_I2C_INSTANCE (I2CDEV_2)
```
Comment (ot0tot) [inline configs/STELLARF4V2/config.h:108]:
```suggestion
#define MAG_I2C_INSTANCE I2CDEV_2
#define BARO_I2C_INSTANCE I2CDEV_2
```

File: configs/STELLARF4V2/config.h [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+#define DASHBOARD_I2C_INSTANCE (I2CDEV_2)
+
+#define DEFAULT_DSHOT_BURST DSHOT_DMAR_ON
```
Comment (ot0tot) [inline configs/STELLARF4V2/config.h:111]: Burst DMA should not be used with F4 MCUs as it is incompatible with bidirectional DShot. 
```suggestion
#define DEFAULT_DSHOT_BURST DSHOT_DMAR_OFF
```

File: configs/STELLARF4V2/config.h [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+                            TIMER_PIN_MAP( 8,  PA9,  1, -1 ) \
+                            TIMER_PIN_MAP( 9,  PA10, 1, -1 ) \
+                            TIMER_PIN_MAP( 10, PA8,  1, -1 ) \
+                            TIMER_PIN_MAP( 11, PA3 , 2,  0 ) \
+                            TIMER_PIN_MAP( 12, PB0,  2, -1 )
+
+#define ADC1_DMA_OPT 0 // DMA2 Stream 0 Channel 0
+
+#define SERIALRX_UART SERIAL_PORT_USART2
+#define SERIALRX_PROVIDER SERIALRX_CRSF
```
Comment (ot0tot) [inline configs/STELLARF4V2/config.h:105]: Does this FC include a receiver? Why include these in the config?

Comment (cvetaevvitaliy) [reply configs/STELLARF4V2/config.h:105]: FC has convenient place for RX

File: configs/STELLARF4V2/config.h (hunk 5 above)
Comment (cvetaevvitaliy) [reply configs/STELLARF4V2/config.h]: refactored to ADC3 DMA2 Stream 1 Channel 2

File: configs/STELLARF4V2/config.h [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+#define MOTOR2_PIN          PC8 // TIM8 CH3 (AF3) DMA2 Stream 4 Channel 7
+#define MOTOR3_PIN          PC7 // TIM8 CH2 (AF3) DMA2 Stream 3 Channel 7
+#define MOTOR4_PIN          PC6 // TIM8 CH1 (AF3) DMA2 Stream 2 Channel 7
+#define SERVO1_PIN          PB6 // TIM4 CH1 (AF2)
+#define SERVO2_PIN          PB7 // TIM4 CH2 (AF2)
+#define SERVO3_PIN          PB8 // TIM4 CH3 (AF2)
+#define SERVO4_PIN          PB9 // TIM4 CH4 (AF2)
+#define SERVO5_PIN          PA9 // TIM1 CH2 (AF1)
+#define SERVO6_PIN          PA10 // TIM1 CH3 (AF1) or UART1_RX_PIN
+#define SERVO7_PIN          PA8 // TIM1 CH1 (AF1) or UART1_TX_PIN
```
Comment (ot0tot) [inline configs/STELLARF4V2/config.h:57]:
```suggestion
#define SERVO5_PIN          PA9 // TIM1 CH2 (AF1) or UART1_TX_PIN
#define SERVO6_PIN          PA10 // TIM1 CH3 (AF1) or UART1_RX_PIN
#define SERVO7_PIN          PA8 // TIM1 CH1 (AF1) 

```

**Review by ot0tot — APPROVED** [review]

**Comment by ot0tot** [comment]
Can also add `#define BOX_USER1_NAME       "CAM 1,2"` to set the name for USER1 in the modes tab.

**Comment by cvetaevvitaliy** [comment]
> Can also add `#define BOX_USER1_NAME       "CAM 1,2"` to set the name for USER1 in the modes tab.

Yes! I forgot, I made this implementation in 4.5.x
Thanks for remember.

**Pull request merged**